### PR TITLE
feat(hr): add new hr component

### PR DESCRIPTION
**Pull request by sjohnson-sage**
### Proposed behaviour

Add new `Hr` component. This is primarily to be used to make a distinct separation between fields in a form but it could have other uses as well.
image

It has 4 props, one for each margin to enable the consumer to display it in alignment with their specific use case. The top and bottom margin props are integer values that will be multiplied by the theme spacing constant (8px). The left and right margins can be any valid CSS margin string in order to make alignment easy for the consumer.

### Current behaviour

Component does not exist.

### Checklist

- [x] Screenshots are included in the PR
- [x] Carbon implementation and Design System documentation are congruent
- [x] All themes are supported
- [x] Commits follow our style guide
- [x] Unit tests added or updated
- <del>[ ] Cypress automation tests added or updated</del>
- [x] Storybook added or updated
- [x] Typescript `d.ts` file added or updated

### Testing instructions
Divider testable at https://codesandbox.io/s/carbon-quickstart-chix8

### CODE REVIEW

File: src/components/divider/divider.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export interface DividerProps {
+  /** Margin top, this value will be multiplied by the theme spacing constant (8) */
+  marginTop: FormFieldSpacing;
+
+  /** Margin bottom, this value will be multiplied by the theme spacing constant (8) */
+  marginBottom: FormFieldSpacing;
+
+  /** Margin left, any valid css value */
+  marginLeft: string;
```
Comment (edleeks87) [inline src/components/divider/divider.d.ts:12]: Should `marginLeft` and `marginRight` not have `?` as they don't have default values and aren't required?

File: src/components/divider/divider.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,84 @@
+
```
Comment (edleeks87) [inline src/components/divider/divider.stories.mdx:1]: Empty line here

File: src/components/divider/divider.d.ts (hunk 1 above)
Comment (sjohnson-sage) [reply src/components/divider/divider.d.ts:12]: Good point yes they should!

File: src/components/divider/divider.component.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+const Divider = ({
+  marginTop = 3,
+  marginBottom = 3,
+  marginLeft,
+  marginRight
+}) => {
+  return (
+    <StyledDivider
+      marginTop={ marginTop }
```
Comment (DlgSHi) [inline src/components/divider/divider.component.js:14]: could we add some `data-component='divider'` here? We will need it further in `cypress` tests

File: src/components/divider/divider.d.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import { FormFieldSpacing } from '../../utils/helpers/options-helper/options-helper';
+
+export interface DividerProps {
+  /** Margin top, this value will be multiplied by the theme spacing constant (8) */
+  marginTop: FormFieldSpacing;
+
```
Comment (DlgSHi) [inline src/components/divider/divider.d.ts:7]: do we need white lines here ?

Comment (sjohnson-sage) [reply src/components/divider/divider.d.ts:7]: All the other components seem to so I just followed that convention, happy to remove though

**Pull request merged**